mdpow/workflows/dihedrals.py

```diff
@@ -462,38 +523,47 @@ def dihedral_violins(df, width=0.9, solvents=SOLVENTS_DEFAULT):
     # number of Coul windows + 1 / number of VDW windows
     # (+1 for additional space with axes)
     width_ratios = [len(df[df['interaction'] == "Coulomb"]["lambda"].unique()) + 1,
-                    len(df[df['interaction'] == "VDW"]["lambda"].unique())]
-
-    solvs = np.asarray(solvents)
-    solv2 = 'octanol'
-    if solvs.size > 1:
-        solv2 = solvs[1]
+                    len(df[df['interaction'] == "VDW"]["lambda"].unique()),
+                    len(df[df['interaction'] == "Coulomb"]["lambda"].unique()) - 1] # was - 1
+
+    # this method allows for one solvent
+    # to be plotted correctly
+    solvs = list(solvents)
+    if len(solvs) < 2:
+        solvs.append('N/A')
```

Comment:
Can we use `None` instead of string `"N/A"`? That would be the canonical Python approach.

Reply:
Yes, if `None` will be displayed on the plot. Currently, two color bars exist in the legend regardless if one or two solvents are plotted, so `N/A` is displayed next to the unused color bar when only one solvent is plotted.

I am sure there is a way to remove the unused color bar from the legend when plotting only one solvent, I just haven't figured it out yet.

Reply:
@orbeckst reminder

Reply:
Can't we change the sns.catplot() to properly deal with the situation? See below...

Reply:
Btw, perhaps add an
```python
assert 0 < len(solvents) < 3
```
just to be sure (instead of a proper input check)